Add back OSX build on travis

apingault opened the pull request:
BEGINRELEASENOTES
- Add OSX build on travis.
  - Only clang, as no root binaries available for gcc.
  - Build last two release of osx (10.12 and 10.13)
  - Builds are allowed to fail to not pollute the travis build history when clang add some fancy new changes to their compiler :)
 - Add `ROOT_VERSION` as env variable in travis.yml
 - Add `FastFinish` option in travis
    - Fails the build as soon as one linux job failed
    - Do not wait for osx build to finish to mark the build as successfull
ENDRELEASENOTES

TODO: 
 - [x] Restore the `WARNING_AS_ERROR` Flags in the cmake configuration once https://github.com/DQM4HEP/dqm4hep-core/pull/55 is merged

Commit:
Add osx build on travis but allow it to fail

Commit:
Trigger travis

Commit:
Fix travis to not update-alternatives on osx

Commit:
Try again

Commit:
Add mysql install on osx

Commit:
Try again mysql

Commit:
Update travis: don't wait for osx build status

Commit:
Update travis for osx: - gcc build + 10.13 clang

No root binaries available for osx+gcc.

Commit:
Update osx_image in travis for current root build

+ Add ROOT_VERSION env var

Commit:
Update install_dep script to dl proper root bins

Commit:
Update build matrix

Commit:
Update build matrix

Commit:
test update osx_image for osx 10.13

Commit:
Fix custom test link problem on osx

Commit:
Fix typo for osx 10.13

rete commented on .travis.yml:
```diff
@@ -14,22 +32,23 @@ addons:
 before_install:
   - ./.travis-ci.d/install_dependencies.sh
+  - if [[ "${TRAVIS_OS_NAME}" == "osx"  ]];
+    then brew install mysql;
```
No TravisCI addons for that ?

apingault replied:
Apparently not: https://docs.travis-ci.com/user/reference/osx/#Homebrew

rete replied:
Fair !

rete commented on source/CMakeLists.txt:
```diff
@@ -49,6 +49,8 @@ if( DQM4HEP_TESTING )
   dqm4hep_add_test( NAME test-xmlparser ARGS "${PROJECT_SOURCE_DIR}/tests/test_xmlparser.xml" )
   
   # Super-custom test
+  set(CUSTOM_TEST_LIBRARY_DIR ${${PROJECT_NAME}_DEPENDS_LIBRARY_DIRS})
```
Maaaaaaad world
https://www.youtube.com/watch?v=4N3N1MlvVc4

apingault replied:
Yeah...took me more than 1hour to find the problem and make this work

Commit:
Add back `ldd` check after compilation

Commit:
Disable running ctests from make on osx

For some reason running ctests from make leads to a `Child aborted` exception being thrown.
Running the tests manually works without any problem though...

apingault committed:
Merge branch 'master' into osxTravis
